tests/simulator/diagnostics.py

```diff
@@ -20,16 +21,25 @@ def dup(dic):
     dic.update({"diags_fn": lambda model: dump_all_diags(model.populations)})
     return dic
 
+@ddt
 class DiagnosticsTest(unittest.TestCase):
 
-    def test_dump_diags_with_killing_dman_1d(self):
+    @data(
+      dup({
+        "smallest_patch_size": 5,
+        "largest_patch_size": 64}),
+      dup({
+        "smallest_patch_size": 65,
+        "largest_patch_size": 65})
+    )
+    def test_dump_diags_with_killing_dman_1d(self, input):
 
         for interp in range(1, 4):
 
             if rank == 0 and os.path.exists(out):
                 shutil.rmtree(out)
 
-            self.dman, self.sim, self.hier = create_simulator(1, interp, **dup({}))
+            self.dman, self.sim, self.hier = create_simulator(1, interp, **input)
```

Comment:
we should probably test that all diags we require are indeed created and not only the EM file. That'd also be useful to have the list of names for diag files somewhere !

Reply:
the filename could be provided by the python diagnostic object, which should be accessible here.
just loop them and check the files exist

Reply:
this fails on particle ghost files, but this may be because of hierarchy from h5 reasons